tests/test_lag_2.py

```diff
@@ -119,8 +211,38 @@ def get_conn_graph_facts(testbed_devices, host):
     result = localhost.conn_graph_facts(host=host, filename=lab_conn_graph_file)['ansible_facts']
     return result
 
+def get_lag_intfs(lag_facts, lag_name):
+    # Figure out interface informations
+    po_interfaces = lag_facts['lags'][lag_name]['po_config']['ports']
+    intf = lag_facts['lags'][lag_name]['po_config']['ports'].keys()[0]
+    return intf, po_interfaces
+
+def check_flap(lag_facts, lag_name):
+    po_intf_num = len(lag_facts['lags'][lag_name]['po_config']['ports'])
+    po_min_links = lag_facts['lags'][lag_name]['po_config']['runner']['min_ports']
+    return ((po_intf_num - 1) * 100 / po_min_links) < 75
+
 def set_interface_lacp_rate(vm_host, intf, mode):
     vm_host.eos_config(
         lines=['lacp rate %s' % mode],
         parents='interface %s' % intf)
-    logging.info("Set interface [%s] lacp rate to [%s]" % (intf, mode))
\ No newline at end of file
+    logging.info("Set interface [%s] lacp rate to [%s]" % (intf, mode))
+
+def set_neighbor_interface(vm_host, neighbor_interface, shut):
+    vm_host.eos_config(
+        lins=['%sshutdown' % ('' if shut else 'no ')],
+        parents='interface %s' % neighbor_interface)
+    logging.info('%s interface [%s]' % ('Shut' if shut else 'No shut', neighbor_interface))
+
+def shell_retry(host, command, retries, delay):
```

Comment:
I am not sure if pytest ansible support retries naturally, then I create this method by myself.
If we need create a more general retry util, we might need allow user to pass a lambda expr to this method to check the output result.

Reply:
There is a wait_until helper function, maybe you can take a look to see if that addresses your needs?